scipy/linalg/tests/test_lapack.py

```diff
@@ -1705,3 +1705,201 @@ def test_getc2_gesc2():
         else:
             assert_array_almost_equal(desired_cplx.astype(dtype),
                                       x/scale, decimal=4)
+
+
+def generate_random_dtype_array(shape, dtype):
+    # generates a random matrix of desired data type of shape
+    if dtype in COMPLEX_DTYPES:
+        return (np.random.rand(*shape)
+                + np.random.rand(*shape)*1.0j).astype(dtype)
+    return np.random.rand(*shape).astype(dtype)
+
+
+@pytest.mark.parametrize("dtype", DTYPES)
+@pytest.mark.parametrize("fact,dl_d_du_lambda",
+                         [("F", lambda dl, d, du: get_lapack_funcs('gttrf',
+                                                                   dtype=d.dtype)(dl, d, du)),
+                          ("N", lambda dl, d, du: (None, None, None, None,
+                                                   None))])
+def test_gtsvx(dtype, fact, dl_d_du_lambda):
+    """
+    This test uses ?gtsvx to solve a random Ax=b system for each dtype.
+    It tests that the outputs define an LU matrix, that inputs are unmodified,
+    transposal options, incompatible shapes, singular matrices, and
+    singular factorizations. It parametrizes DTYPES and the fact value along
+    with the fact related inputs.
+    """
+    np.random.seed(42)
+    # set test tolerance appropriate for dtype
+    rtol = 250 * np.finfo(dtype).eps
+    atol = 100 * np.finfo(dtype).eps
+    # obtain routine
+    gtsvx = get_lapack_funcs('gtsvx', dtype=dtype)
```

Comment:
```suggestion
    gtsvx = get_lapack_funcs('gtsvx', dtype=dtype)[0]
```
I think it returns a list.

Reply:
It seems to be working for me without that, for me it is returning a fortran object.

I'll have to commit changes tomorrow morning, as I've got to go to my 7 am tomorrow. Sorry that this wasn't fully finished by today.

Reply:
@mdhaber this is fine. `get_lapack_funcs` returns the fortran function directly if `names` is a string:
```python
Python 3.7.4 (default, Aug  9 2019, 18:34:13) [MSC v.1915 64 bit (AMD64)]
Type 'copyright', 'credits' or 'license' for more information
IPython 7.12.0 -- An enhanced Interactive Python. Type '?' for help.

In [1]: import numpy as np
In [2]: from scipy.linalg import get_lapack_funcs
In [3]: get_lapack_funcs('gtsv', dtype=float)
Out[3]: <fortran object>
In [4]: get_lapack_funcs(('gtsv',), dtype=float)
Out[4]: [<fortran object>]
```

Reply:
Ah.